File: backend/Project1.Api/Entities/SupplierProduct.cs
namespace Project1.Api.Entities;

public sealed class SupplierProduct
{
    public int Id { get; set; }

    public int SupplierId { get; set; }

    public Supplier Supplier { get; set; } = null!;

    public int ProductId { get; set; }

    public Product Product { get; set; } = null!;

    public bool IsPreferred { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTimeOffset CreatedAtUtc { get; set; } = DateTimeOffset.UtcNow;

    public DateTimeOffset? UpdatedAtUtc { get; set; }
}

File: frontend/src/types/supplierProduct.ts
export interface SupplierProduct {
  id: number
  supplierId: number
  supplierCode: string
  supplierName: string
  productId: number
  productCode: string
  productName: string
  unitOfMeasureCode: string
  unitOfMeasureName: string
  productDefaultUnitPrice: number
  isPreferred: boolean
  isActive: boolean
  createdAtUtc: string
  updatedAtUtc: string | null
}

export interface CreateSupplierProductRequest {
  supplierId: number
  productId: number
  isPreferred: boolean
}

export interface UpdateSupplierProductRequest {
  isPreferred: boolean
  isActive: boolean
}

export interface SupplierProductFormValues extends CreateSupplierProductRequest {
  isActive: boolean
}

export interface SupplierProductFilters {
  supplierId?: number
  productId?: number
  includeInactive?: boolean
}
